### PR TITLE
add database documentation

### PR DESCRIPTION
**Pull request by ktdreyer**
Document how to set up a local Postgres environment, along with some information about the production environment.

### CODE REVIEW

File: database.rst [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+Note: the ``tree_packages`` table is 1.3GB compressed (unknown size
+uncompressed). It took 18 minutes to dump from prod over my VPN connection
+(1.25MB/s).
+
+I've not yet been able to restore this ``tree_packages`` table to a development
+VM. On RHEL 5, ``pg_restore`` churns for an hour and then OOMs even with 4GB
+RAM. On RHEL 7 ``pg_restore`` does not leak memory, but I kept running out of
+disk space in ``/var/lib/pgsql/data``. It took up 50GB of space before I gave
+up.
```
Comment (hluk) [inline database.rst]: Woah, that's crazy.

Comment (ktdreyer) [reply database.rst]: Yeah, I need to retry - I'm still shuffling stuff around on my laptop so I have enough space :cry:

Comment (hluk) [reply database.rst]: Just did today: `docker image prune && docker volume prune`

It saved me few GiBs.

**Comment by ktdreyer** [comment]
@tkopecek FYI

File: database.rst (hunk 1 above)
Comment (ktdreyer) [reply database.rst]: Yeah, I still cannot import this table with `pg_restore`, even with 6GB RAM and 4vCPUs, and tuning postgres to use less memory. Are you ok merging this as-is, and we'll come up with some better solution later?

Comment (hluk) [reply database.rst]: Sure, let's merge it.

**Pull request merged**